community/kernel/src/main/java/org/neo4j/kernel/impl/core/NodeProxy.java

```diff
@@ -746,14 +742,29 @@ public int getDegree( RelationshipType type, Direction direction )
     @Override
     public Iterable<RelationshipType> getRelationshipTypes()
     {
-        try ( Statement statement = spi.statement() )
+        KernelTransaction transaction = safeAcquireTransaction();
+        try ( RelationshipTraversalCursor relationships = transaction.cursors().allocateRelationshipTraversalCursor();
+              Statement ignore = transaction.acquireStatement() )
         {
-            PrimitiveIntIterator relTypes = statement.readOperations().nodeGetRelationshipTypes( nodeId );
-            return asList( map( relTypeId -> convertToRelationshipType( statement, relTypeId ), relTypes ) );
+            NodeCursor nodes = transaction.nodeCursor();
+            TokenRead tokenRead = transaction.tokenRead();
+            singleNode( transaction, nodes );
+            if ( !nodes.next() )
+            {
+                throw new NotFoundException( "Node not found." );
+            }
+            nodes.allRelationships( relationships );
+            List<RelationshipType> types = new ArrayList<>();
+            while ( relationships.next() )
+            {
+                types.add( RelationshipType.withName( tokenRead.relationshipTypeName( relationships.label() ) ) );
+            }
```

Comment:
The previous implementation only returned each type once, but this one will return as many relationship types as the node has relationships.